third_party/accessibility/ax/platform/ax_platform_node_textrangeprovider_win.h

```diff
@@ -0,0 +1,283 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef UI_ACCESSIBILITY_PLATFORM_AX_PLATFORM_NODE_TEXTRANGEPROVIDER_WIN_H_
+#define UI_ACCESSIBILITY_PLATFORM_AX_PLATFORM_NODE_TEXTRANGEPROVIDER_WIN_H_
+
+#include <atlbase.h>
+#include <atlcom.h>
+
+#include <UIAutomationCore.h>
+
+#include "ax/ax_node_position.h"
+#include "ax/ax_tree_observer.h"
+#include "ax/platform/ax_platform_node_delegate.h"
+#include "ax/platform/ax_platform_node_win.h"
+
+namespace ui {
+
+class AX_EXPORT __declspec(uuid("3071e40d-a10d-45ff-a59f-6e8e1138e2c1"))
+    AXPlatformNodeTextRangeProviderWin
+    : public CComObjectRootEx<CComMultiThreadModel>,
+      public ITextRangeProvider {
+ public:
+  BEGIN_COM_MAP(AXPlatformNodeTextRangeProviderWin)
+  COM_INTERFACE_ENTRY(ITextRangeProvider)
+  COM_INTERFACE_ENTRY(AXPlatformNodeTextRangeProviderWin)
+  END_COM_MAP()
+
+  AXPlatformNodeTextRangeProviderWin();
+  ~AXPlatformNodeTextRangeProviderWin();
+
+  static ITextRangeProvider* CreateTextRangeProvider(
+      AXNodePosition::AXPositionInstance start,
+      AXNodePosition::AXPositionInstance end);
+
+  static ITextRangeProvider* CreateTextRangeProviderForTesting(
```

Comment:
Can we keep Chromium's comment here?

```suggestion
  // Creates an instance of the class for unit tests, where AXPlatformNodes	
  // cannot be queried automatically from endpoints.
  static ITextRangeProvider* CreateTextRangeProviderForTesting(
```